Updated GitHub Actions documentation

sudomateo opened the pull request:
Updated the GitHub Actions documentation to refer to the new `hashicorp/setup-terraform` GitHub action located [here](https://github.com/hashicorp/setup-terraform).

The old `hashicorp/terraform-github-actions` GitHub action located [here](https://github.com/hashicorp/terraform-github-actions) action will be marked as read-only and no longer be maintained as of next week.

nfagerlund commented on content/redirects.txt:
```diff
@@ -422,17 +422,8 @@
-/docs/github-actions/directories.html                               /docs/github-actions/common-tasks/working-directory.html
-/docs/github-actions/remote-backend.html                            /docs/github-actions/common-tasks/backends.html
-/docs/github-actions/terraform-versions.html                        /docs/github-actions/common-tasks/terraform-versions.html
-/docs/github-actions/workspaces.html                                /docs/github-actions/common-tasks/workspaces.html
```
Ok, yeah: We need to preserve all the existing redirects, and add new redirects for the pages this PR deletes.

sudomateo replied:
Added back the redirects and redirected them to the new index.

sudomateo commented on content/source/docs/github-actions/index.html.md:
```diff
@@ -1,28 +1,10 @@
 # Terraform GitHub Actions
 
-This is the documentation for Terraform GitHub Actions.
+[GitHub Actions](https://help.github.com/actions) allow you to execute one or more workflows in response to GitHub events such as a pushing a commit or opening a pull request. Through GitHub Actions, a user can automate common Terraform workflows such as installing Terraform CLI and executing various `terraform` commands. HashiCorp maintains the following GitHub Actions to help automate Terraform ...
```
Added a note addressing your "BUT" point. Let me know what you think.

sudomateo commented on content/source/docs/github-actions/setup-terraform.html.md:
```diff
@@ -0,0 +1,128 @@
+
+* Downloading a specific version of Terraform CLI and adding it to the `PATH`.
+* Configuring the [Terraform CLI configuration file](/docs/commands/cli-config.html) with a Terraform Cloud/Enterprise hostname and API token.
+* Optionally installing a wrapper script to wrap subsequent calls of the `terraform` binary and expose its STDOUT, STDERR, and exit code as outputs named `stdout`, `stderr`, and `exitcode` respectively.
```
Added this!

sudomateo commented on content/source/docs/github-actions/setup-terraform.html.md:
```diff
@@ -0,0 +1,128 @@
+
+The source code and reference documentation for the `hashicorp/setup-terraform` action can be found [here](https://github.com/hashicorp/setup-terraform).
+
+Let's see how to use this action in a workflow!
```
I agree regarding the Learn guide! We're going to move forward with a stripped down version of this content for the website and retroactively come back and add the Learn guide.

sudomateo commented on content/source/docs/github-actions/setup-terraform.html.md:
```diff
@@ -0,0 +1,128 @@
+* Downloading a specific version of Terraform CLI and adding it to the `PATH`.
+* Configuring the [Terraform CLI configuration file](/docs/commands/cli-config.html) with a Terraform Cloud/Enterprise hostname and API token.
+* Optionally installing a wrapper script to wrap subsequent calls of the `terraform` binary and expose its STDOUT, STDERR, and exit code as outputs named `stdout`, `stderr`, and `exitcode` respectively.
+
```
Done!

sudomateo commented on content/source/docs/github-actions/index.html.md:
```diff
@@ -1,28 +1,14 @@
+[GitHub Actions](https://help.github.com/actions) allow you to execute one or more workflows in response to GitHub events such as a pushing a commit or opening a pull request. Through GitHub Actions, a user can automate common Terraform workflows such as installing Terraform CLI and executing various `terraform` commands.
 
-[GitHub Actions](https://developer.github.com/actions) allow you to run commands in response to GitHub events such as pushing a new commit. Terraform GitHub Actions are designed to execute Terraform commands to help you review and validate Terraform changes. When Terraform GitHub Actions are executed on a pull request, a comment may be posted on the pull request conversation detailing the outcome ...
+-> **Note** The former [hashicorp/terraform-github-actions](https://github.com/hashicorp/terraform-github-actions/) GitHub action is no longer maintained. It has been superseded by the `hashicorp/setup-terraform` GitHub action listed below. Although no functionality has been lost, workflow files will need to be reformatted to in order to use the `hashicorp/setup-terraform` GitHub action.
```
Here is where I added the note for users of the old actions.

nfagerlund replied:
👍🏼 Perfect.

sudomateo commented:
@nfagerlund I addressed all the points we chatted about. Here's what I did.

- Added back the redirects and redirected them to the new index.
- Added a note for users of the older actions.
- Adjusted the verbiage slightly in a few places.

Let me know what you think!
